kpops/components/base_components/pipeline_component.py

```diff
@@ -64,6 +71,25 @@ def __init__(self, **kwargs) -> None:
     def full_name(self) -> str:
         return self.prefix + self.name
 
+    @classmethod
+    def get_parents(
+        cls: type[Self], __class_or_tuple: type = BaseDefaultsComponent
+    ) -> list[str]:
```

Comment:
I don't really like that this is just a list of strings (loss of information). it would be better to return a sequence of classes, then the caller can choose to do `cls.type` if necessary